subprojects/core/src/main/java/org/gradle/process/internal/worker/DefaultWorkerProcess.java

```diff
@@ -66,6 +66,10 @@ public JvmMemoryStatus getJvmMemoryStatus() {
 
     @Override
     public void stopNow() {
+        if (connection != null) {
```

Comment:
That doesn't seem thread-save. Also, what if the `onConnect` call comes shortly after the `abort` call?

Reply:
Moved this into a `try-finally` block using the `lock`.

As for `onConnect` after `abort`, do you think it would suffice to `abort` that `ObjectConnection`?  Or do you foresee any problems with that?

Reply:
I think that should do.